--- a/src/main/java/com/docuseal/AsyncRawDocusealClient.java
+++ b/src/main/java/com/docuseal/AsyncRawDocusealClient.java
@@ -2149,7 +2149,8 @@ public class AsyncRawDocusealClient {
         HttpUrl.Builder httpUrl = HttpUrl.parse(this.clientOptions.environment().getUrl())
                 .newBuilder()
                 .addPathSegments("templates")
-                .addPathSegment(Integer.toString(id) + "?permanently=true");
+                .addPathSegment(Integer.toString(id))
+                .addQueryParameter("permanently", "true");
         if (requestOptions != null) {
             requestOptions.getQueryParameters().forEach((_key, _value) -> {
                 httpUrl.addQueryParameter(_key, _value);
@@ -2243,7 +2244,8 @@ public class AsyncRawDocusealClient {
         HttpUrl.Builder httpUrl = HttpUrl.parse(this.clientOptions.environment().getUrl())
                 .newBuilder()
                 .addPathSegments("submissions")
-                .addPathSegment(Integer.toString(id) + "?permanently=true");
+                .addPathSegment(Integer.toString(id))
+                .addQueryParameter("permanently", "true");
         if (requestOptions != null) {
             requestOptions.getQueryParameters().forEach((_key, _value) -> {
                 httpUrl.addQueryParameter(_key, _value);
--- a/src/main/java/com/docuseal/RawDocusealClient.java
+++ b/src/main/java/com/docuseal/RawDocusealClient.java
@@ -1786,7 +1786,8 @@ public class RawDocusealClient {
         HttpUrl.Builder httpUrl = HttpUrl.parse(this.clientOptions.environment().getUrl())
                 .newBuilder()
                 .addPathSegments("templates")
-                .addPathSegment(Integer.toString(id) + "?permanently=true");
+                .addPathSegment(Integer.toString(id))
+                .addQueryParameter("permanently", "true");
         if (requestOptions != null) {
             requestOptions.getQueryParameters().forEach((_key, _value) -> {
                 httpUrl.addQueryParameter(_key, _value);
@@ -1862,7 +1863,8 @@ public class RawDocusealClient {
         HttpUrl.Builder httpUrl = HttpUrl.parse(this.clientOptions.environment().getUrl())
                 .newBuilder()
                 .addPathSegments("submissions")
-                .addPathSegment(Integer.toString(id) + "?permanently=true");
+                .addPathSegment(Integer.toString(id))
+                .addQueryParameter("permanently", "true");
         if (requestOptions != null) {
             requestOptions.getQueryParameters().forEach((_key, _value) -> {
                 httpUrl.addQueryParameter(_key, _value);
