--- a/src/main/java/com/docuseal/core/ClientOptions.java
+++ b/src/main/java/com/docuseal/core/ClientOptions.java
@@ -45,11 +45,6 @@ public final class ClientOptions {
         this.environment = environment;
         this.headers = new HashMap<>();
         this.headers.putAll(headers);
-        this.headers.putAll(new HashMap<String, String>() {
-            {
-                put("X-Fern-Language", "JAVA");
-            }
-        });
         this.headerSuppliers = headerSuppliers;
         this.httpClient = httpClient;
         this.timeout = timeout;
